data/src/languages_update.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import collections
+import json
+import operator
+import os
+import sys
+
+import unicodedataplus
+from data.src.codes import LANGUAGES_PATH, TSV_DIRECTORY_PATH
+
+from typing import Dict, Tuple, Union
+
+def _detect_best_script_name(
+    word: str, strict: bool = True
+) -> Union[Tuple[str, float], None]:
+    """Returns the most likely script name (rather than ISO 15924 code) the
+    word belongs to along with the corresponding confidence expressed as a
+    maximum likelihood estimate computed over the `word` sample. If `strict`
+    is enabled, then all the characters must belong to the same script and
+    `(None, None)` is returned on failure.
+
+    Example: "ژۇرنال" -> ("Arabic", 1.0).
+    """
+    script_counts: Dict[str, float] = collections.defaultdict(float)
```

Comment:
Use this more specific type instead: https://docs.python.org/3/library/typing.html#typing.DefaultDict